src/custom/state/claim/actions.ts

```diff
@@ -12,14 +13,20 @@ export type ClaimActions = {
   setInputAddress: (payload: string) => void
   setActiveClaimAccount: (payload: string) => void
   setActiveClaimAccountENS: (payload: string) => void
+
   // search
   setIsSearchUsed: (payload: boolean) => void
+
   // claiming
   setClaimStatus: (payload: ClaimStatus) => void
   setClaimedAmount: (payload: number) => void
+
   // investing
   setIsInvestFlowActive: (payload: boolean) => void
   setInvestFlowStep: (payload: number) => void
+  initInvestFlowData: (payload: EnhancedUserClaimData[]) => void
```

Comment:
dont save complex shapes to redux state, it's unoptimised, and we can't debug them in redux dev tools.

in this case `EnhancedUserClaimData` is:
```ts
type EnhancedUserClaimData = UserClaimData & {
  currencyAmount?: CurrencyAmount<Token | GpEther> | undefined
  claimAmount?: CurrencyAmount<Token> | undefined
  price?: Price<Currency, Currency> | undefined
  cost?: CurrencyAmount<Currency> | undefined
  isFree: boolean
}

do we really need all of that? if not only save what we need and only save the string representations of what we need. i realy dont know if we do need this in state tho...

happy to hear otherwise
```

Reply:
Updated with simple redux data structure with just `index` and `investAmount`